include/alpaka/kernel/Traits.hpp

```diff
@@ -83,6 +85,88 @@ namespace alpaka
+            //! When no schedule is defined, return a default one.
+            template<class TKernel, class = void>
+            struct GetOmpSchedule
+            {
+                ALPAKA_FN_HOST static auto get()
+                {
+                    return alpaka::omp::Schedule{};
+                }
+            };
+
+            //! Functor to get OpenMP schedule for kernel classes with
+            //! ompSchedule static member.
+            //! That member is never odr-used by alpaka.
+            template<class TKernel>
+            struct GetOmpSchedule<TKernel, meta::Void<decltype(TKernel::ompSchedule)>>
+            {
+                ALPAKA_FN_HOST static auto get()
+                {
+                    // Just having return TKernel::ompSchedule here would be
+                    // a non-odr use of that variable, since it would be an
+                    // argument of the copy constructor. So have to manually
+                    // create a new identical object and then return it.
+                    return alpaka::omp::Schedule{TKernel::ompSchedule.kind, TKernel::ompSchedule.chunkSize};
+                }
+            };
+        } // namespace detail
+
+        //#############################################################################
+        //! The trait for getting the schedule to use when a kernel is run using
+        //! the CpuOmp2Blocks accelerator.
+        //!
+        //! Has no effect on other accelerators or when run using OpenMP
+        //! implementation not supporting at least version 3.0.
+        //!
+        //! A user could either specialize this trait for their kernel, or define
+        //! a public static member ompSchedule of type alpaka::omp::Schedule
+        //! inside it, which would be picked up by this implementation.
+        //! In the latter case, alpaka never odr-uses that member.
+        //!
+        //! \tparam TKernelFnObj The kernel function object.
+        //! \tparam TAcc The accelerator.
+        //!
+        //! The default implementation returns 0.
+        template<typename TKernelFnObj, typename TAcc, typename TSfinae = void>
+        struct OmpSchedule
+        {
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored                                                                                  \
+        "-Wdocumentation" // clang does not support the syntax for variadic template arguments "args,..."
+#endif
+            //-----------------------------------------------------------------------------
+            //! \param kernelFnObj The kernel object for which the schedule should be returned.
+            //! \param blockThreadExtent The block thread extent.
+            //! \param threadElemExtent The thread element extent.
+            //! \tparam TArgs The kernel invocation argument types pack.
+            //! \param args,... The kernel invocation arguments.
```

Comment:
is it not possible to write only `args`. The macros around the comments make it hard to read the file

Reply:
I agree. This pattern is also present in [other places](https://github.com/alpaka-group/alpaka/blob/a8e26171e86b65754e66f5f3b9f9ef0ac23729d6/include/alpaka/kernel/Traits.hpp#L54-L69) where I copied it from. I will make another PR that implements your suggestion. Not only it makes the code more readable, it is also more technically accurate, as `args` here is all arguments already and `args...` is just pack extension